builtin/logical/database/dbplugin/plugin.go

```diff
@@ -0,0 +1,138 @@
+package dbplugin
+
+import (
+	"fmt"
+	"net/rpc"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/pluginutil"
+	log "github.com/mgutz/logxi/v1"
+)
+
+// DatabaseType is the interface that all database objects must implement.
+type DatabaseType interface {
```

Comment:
Nitpick, but might this be better named `Database`? It's not really a type per se, it's more a session, or client, or instantiation.